src/main/java/com/frameworkium/core/ui/pages/BasePage.java

```diff
@@ -100,15 +97,14 @@ public T get(long timeout) {
     @SuppressWarnings("unchecked")
     public T get() {
 
+        //Populate Page Object
         HtmlElementLoader.populatePageObject(this, driver);
 
-        // wait for page to load
-        if (isPageAngularJS()) {
-            waitForAngularRequestsToFinish();
-        }
+        //Wait for Elements & JS
         visibility.waitForAnnotatedElementVisibility(this);
+        javascriptWait.waitForJavascriptEventsOnLoad();
```

Comment:
Makes sense. Just checking as it changed.

I thought waiting for JS Framework stuff before checking elements would be better if JS changes the elements? But if it doesn't work then that's ok as it would still wait for elements created by JS first such that the wait for JS would run after but wouldn't need to wait. If there were no waits for elements which requires JS then this would still work as expected.